src/main/java/utils/StringToDate.java

```
@@ -0,0 +1,141 @@
+package utils;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * This class returns literal dates converted to date format.
+ */
+public class StringToDate {
+
+    SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+
+    /**
+     * Converts literal date String to a date format.
+     *
+     * @param stringToConvert a literal date String.
+     * @return a date format.
+     */
+    public Date convert(final String stringToConvert) {
+        if (stringToConvert == null) {
+            throw new IllegalArgumentException("Value is null");
+        }
+        if (stringToConvert.equals("")) {
+            throw new IllegalArgumentException("Value is empty");
+        }
+
+        if (stringToConvert.matches("TODAY|TOMORROW|YESTERDAY")) {
+            return dayName(stringToConvert);
+        }
+
+        if (stringToConvert.contains("ago") || stringToConvert.contains("from now")) {
+            return getDateValue(stringToConvert);
+        }
+
+        if (stringToConvert.contains("/")) {
+            try {
+                return dateFormat.parse(stringToConvert);
+            } catch (ParseException exception) {
+                throw new IllegalArgumentException(exception);
+            }
+        }
+
+        return null;
```

maybe instead of returning null you could return an exception

---

Thanks. I just have done that